sources/web/datalab/polymer/modules/bigquery-file-manager/bigquery-file-manager.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+class BigQueryMethodNotYetImplemented extends Error {
+  constructor(apiFunctionName: string) {
+    super('BigQuery method ' + apiFunctionName + ' is not implemented');
+  }
+}
+
+/**
+ * A file manager that wraps the BigQuery API so that we can browse BQ projects,
+ * datasets, and tables like a filesystem.
+ */
+class BigQueryFileManager implements FileManager {
+
+  /**
```

Comment:
Can we remove these comments? They're not related to BigQuery. And I think we should keep these comments on the interface only anyway, we can add BigQueryFileManager related comments inside the methods.